src/context/context.js

```diff
@@ -0,0 +1,40 @@
+import React, {Component} from 'react'
+import {db} from '../firebase.js'
+
+const UserContext = React.createContext()
+
+class UserProvider extends Component {
```

Comment:
I might suggest using a functional component here rather than a class component. Functional components are able to utilize React hooks which are the future of React. They also allow you to pass state around via props which can be super useful and clean way to build your apps. Here is a quick [high-level on the differences](https://medium.com/@Zwenza/functional-vs-class-components-in-react-231e3fbd7108), but I strongly suggest checking out the [React docs here as well](https://reactjs.org/docs/components-and-props.html#function-and-class-components).